### PR TITLE
Add option to disable auto-focus

### LINKED ISSUE #523: Add a preference to prevent pages from stealing Vimium's focus when they first load
I like to scroll through my tabs using J,K. This works fine, until a tab has a highlighted text area, which the browser starts typing into. Next, I have to press Esc and continue with J,K until i encounter the next highlighted text area.
If wanted to type into a text area, I would select it myself using the f key.
Please fix :)

### PR DESCRIPTION
**Pull request by mrmr1993**
This fixes #523.

### CODE REVIEW

**Comment by smblott-github** [comment]
Hi, @mrmr1993.  I was trying this out on [this site](https://www.google.ie/).  Seems to work when the page is first visited (e.g. in a new tab) and when the page is refreshed with `r`.  Unfortunately, though, not when the page is refreshed with `Ctrl-R`.  Any ideas?

**Comment by mrmr1993** [comment]
@smblott-github is this behaviour the same on other websites with autofocus (eg. [Bing](http://www.bing.com/))?

At a guess, I'd say the new page catches the `keyup` event with `Ctrl-R`, but our round-trip to the background page with `r` takes enough time that the old page gets it instead. If this is the case, dropping `"keyup"` from `allowFocusAfterEvents` should do the trick.

There are some problems with the code that I've noticed while reading it, I'll add a few comments to the code page.

File: content_scripts/vimium_frontend.coffee [hunk 1]
```diff
@@ -162,6 +164,46 @@ initializeWhenEnabled = (newPassKeys) ->
...
+    true
+  unfocus: (event) ->
+    return true unless isFocusable event.target
+    FocusPrevention.lastFocused = event.target
+    DomUtils.suppressPropagation event if FocusPrevention.tryUnfocus()
+
+  # Either add or remove the event listeners responsible for stopping autofocus, depending on add.
+  modifyListeners: (add) ->
+    changeEventListener = if add then "addEventListener" else "removeEventListener"
+    document[changeEventListener] "focus", @unfocus, true
```
Comment (mrmr1993) [inline content_scripts/vimium_frontend.coffee]: Should have a `blur` listener that does `DomUtils.suppressPropagation event` so the page doesn't see an asymmetry between `blur`s and `focus`es.

Also, the `focus` and `blur` listeners should set `@isStillFocused` to `true` and `false` respectively.

**Comment by smblott-github** [comment]
> is this behaviour the same on other websites with autofocus (eg. Bing)?

I tried bing.  with both `r` and `Ctrl-R`, sometimes it works, sometimes it doesn't.  So, looks like a race condition.

**Comment by smblott-github** [comment]
Closing.  Fixed in #1480.

**Pull request closed without merging**